gdal/frmts/netcdf/netcdfdataset.cpp

```diff
@@ -10963,3 +11119,34 @@ static CPLErr NCDFGetCoordAndBoundVarFullNames( int nCdfId,
 
     return CE_None;
 }
+
+//Check if give type is user defined
+static bool NCDFIsUserDefinedType(int ncid, int type)
+{
+    //Adapted from OPENDAP netcdf_handler
+    //To circumvent use of NC_FIRSTUSERTYPEID
+    //Which is not a part of netcdf 4.1.1 installed on RH
+    //In all later version, type >= NC_FIRSTUSERTYPEID works
+#if NETCDF_HAS_NC4
+    int ntypes;
+    int typeids[NC_MAX_VARS];  // It's likely safe to assume there are
+    int rootid; //For nested datasets
+
+    ERR_RET(NCDFGetRootGroup(ncid, &rootid));
```

Comment:
c:\projects\gdal\gdal\frmts\netcdf\netcdfdataset.cpp(11135): warning C4800: 'CPLErr': forcing value to bool 'true' or 'false' (performance warning)

ERR_RET returns a CPLErr whereas this method returns bool. You should do the error checking manually here